dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Axiom.java

```diff
@@ -80,7 +82,7 @@ public void setName(String name) {
 
     public abstract TupleSet getEncodeTupleSet();
 
-    public abstract BooleanFormula consistent(SolverContext ctx);
+    public abstract BooleanFormula consistent(Set<Tuple> toBeEncoded, Context analysisContext, SolverContext ctx);
```

Comment:
I think the `toBeEncoded` parameter is weird, but so is our current solution :).
Eventually, this parameter will go, because there is simply no other entity in the code base that should change this parameter. Conceptually, it should always coincide with the return value of `getEncodeTupleSet` (right now it doesn't, cause our current code is conceptually wrong).

Reply:
This is not the final signature, but it will ease the change later, when encode sets get moved.